errors.go

```diff
@@ -55,17 +55,19 @@ func isResponseError(req *http.Request, resp *http.Response) *ResponseError {
 		return nil
 	}
 
-	// 429 is returned by standby instances for /sys/health requests and should
-	// not be treated as an error; for other paths the status code indicates
-	// that the quota limit has been reached.
-	if resp.StatusCode == http.StatusTooManyRequests /* 429 */ && req.URL.Path == "/v1/sys/health" {
-		return nil
-	}
-
-	// 472 & 473 are returned by (performance/secondary) standby instances for /sys/health requests and should
-	// not be treated as an error
-	if (resp.StatusCode == 472 || resp.StatusCode == 473) && req.URL.Path == "/v1/sys/health" {
-		return nil
+	// /v1/sys/health returns a few special 4xx status codes that should not be
+	// treated as errors:
+	//
+	//  - 429 if unsealed and standby
+	//  - 472 if disaster recovery mode replication secondary and active
+	//  - 473 if performance standby
+	//
+	// See: https://developer.hashicorp.com/vault/api-docs/system/health
+	if req.URL.Path == "/v1/sys/health" {
+		switch resp.StatusCode {
+		case 429, 472, 473:
+			return nil
+		}
 	}
```

Comment:
```suggestion
	// /v1/sys/health returns a few special status codes 
	// however, the returned response has valued information
	// See: https://developer.hashicorp.com/vault/api-docs/system/health
	if req.URL.Path == "/v1/sys/health" && resp != nil  {
		return nil
	}
```

Reply:
https://github.com/hashicorp/vault/blob/main/api/sys_health.go#L30

Reply:
Ah interesting, in the old library we forced the response codes to be `299`. I wonder if it's really necessary. Let me test it internally...

Reply:
Thanks, @yanjunding!

https://github.com/hashicorp/vault-client-go/pull/221/commits/0d22c5e03c2d90b1458e69a57c51e08f36504fe1